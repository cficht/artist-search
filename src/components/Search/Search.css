.Search {
  display: flex;
  justify-content: center;  
  align-items: center;
  flex-direction: column;

  background-color: red;
}

.Input {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  width: 30%;
  margin: 20px 20px 10px 20px;
  padding: 20px;
  border-radius: 10px;

  background-color: darkblue;

  form {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
  }
  
  input[type=text] {
    margin: 5px;

    font-size: 16px;
  }

  button {
    margin: 15px 5px 5px 5px;

    background-color: #142850;
    color: white;

    font-size: 16px;
    text-align: center;
  }
}

.List {
  display: flex;
  justify-content: center;

  width: 30%;
  margin: 10px 20px 20px 20px;
  padding: 20px;
  border-radius: 10px;

  background-color: darkblue;

  ul {
    margin-block-start: 0em;
    margin-block-end: 0em;
    padding-inline-start: 0px;
    list-style-type: none;
  }

  a {
    color: white;
    font-size: 16px;
  }
}