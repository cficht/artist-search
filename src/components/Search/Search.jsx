import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Search = ({ artistText, artists, handleChange, handleSubmit }) => {
  const artistNodes = artists.map(artist => (
    <li key={artist.id}>
      <Link to={`/artist/${artist.name}/${artist.id}/0`}>{artist.name}</Link>
    </li>
  ));

  return (
    <>
      <section>
        <form onSubmit={handleSubmit}>
          <input type="text" value={artistText || ''} onChange={handleChange}/>
          <button type="submit">Submit</button>
        </form>
      </section>
      <section>
        <ul>
          {artistNodes}
        </ul>
      </section>
    </>
  );
};

Search.propTypes = {
  artistText: PropTypes.string,
  artists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  })).isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Search;
