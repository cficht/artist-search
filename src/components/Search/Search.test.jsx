import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search component', () => {
  it('renders Search', () => {
    const wrapper = shallow(<Search 
      artistText="The Beatles"
      artists={[]}
      handleChange={()=>{}}
      handleSubmit={()=>{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
