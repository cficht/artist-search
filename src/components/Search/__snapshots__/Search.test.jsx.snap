// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<Fragment>
  <section>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="The Beatles"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
  <section>
    <ul />
  </section>
</Fragment>
`;
